compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -2442,6 +2444,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             return Ok(None);
         }
 
+        if !tcx.features().inherent_associated_types {
+            tcx.sess
+                .delay_span_bug(span, "found inherent assoc type without the feature being gated");
```

Comment:
Not using `assert!` or similar as we'd ice when sb. defines & uses an IAT w/ enabling the feature.
Instead of `delay_span_bug`, I could exit early at the very start of the function but that would lead to "spurious" extra errors on IAT use-sites (*ambiguous assoc ty*) next to the feature gate error.